llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -21553,6 +21553,56 @@ bool DAGCombiner::tryStoreMergeOfLoads(SmallVectorImpl<MemOpLink> &StoreNodes,
       JointMemOpVT = EVT::getIntegerVT(Context, SizeInBits);
     }
 
+    auto HasCallInLdStChain = [](SmallVectorImpl<MemOpLink> &StoreNodes,
```

Comment:
Can we pull this out into a static function to make it similar to the `checkMergeStoreCandidatesForDependencies` check?

Reply:
Sure, do you want a static function or a member function like `checkMergeStoreCandidatesForDependencies`? 

Just asking because  `MemOpLink` is a private struct inside the `DAGCombiner` class, so we would need to make it public to pass it as argument of the static function.

Reply:
Oh I didn't realise it was a member function, but yeah a member function sounds fine